src/esp/assets/PTexMeshData.cpp

```diff
@@ -91,6 +91,22 @@ const std::vector<PTexMeshData::MeshData>& PTexMeshData::meshes() const {
 std::string PTexMeshData::atlasFolder() const {
   return atlasFolder_;
 }
+// this is to break the quad into 2 triangles
+// we need this triangle mesh to do object picking
+void computeTriangleMeshIndices(uint64_t numFaces,
```

Comment:
This method looks really useful.  Is there some centralized place where we could put a method like this so that it might be used by other processes/classes?

Reply:
Oh! Can I steal this? :grin: 

I wanted to put something similar directly into Magnum's `MeshTools`, with the added feature of [properly triangulating non-convex quads](https://stackoverflow.com/questions/12239876/fastest-way-of-converting-a-quad-to-a-triangle-strip), and then use it in all places where needed (STL and OBJ meshes, mainly). Extremely hasty ASCII art to visualize what I mean:

```
+                              +
|\                             |\
| \                            | \  
|  +---+           this        |  +---+
|     /                        |  .  /
|    /              --->       | .  /
|   /                          | . /
|  /                           |. /
| /                            |./
|/                             |/
+                              +

                               +
                               |\ .
                               | \  .
                  instead      |  +---+
                  of this      |     /
                               |    /
                   -x->        |   /
                               |  /
                               | /
                               |/
                               + 
```

If you say yes, I can append this to #740 (or open a new PR, whichever you prefer) by tomorrow to avoid blocking you.

Reply:
Be my guest. 
Use it anywhere you want. :)

Reply:
Ok, if you want to use a general-purpose utility, it's available in Magnum as [MeshTools::generateQuadIndices()](https://doc.magnum.graphics/magnum/namespaceMagnum_1_1MeshTools.html#aeba1cfbb6e33040f3025872a0b340db0) since mosra/magnum@b77eafb9deba8228214cd7759d12bbdff555eba0. Simply pull the magnum submodule, nothing else needs to be updated as there was no breaking change since last time.

![image](https://user-images.githubusercontent.com/344828/90986148-e04def80-e580-11ea-9d73-2135df6d1ccc.png)